src/common/libflux/future.c

```diff
@@ -108,6 +108,8 @@ static int now_context_set_timeout (struct now_context *now,
                                     double timeout, void *arg)
 {
     if (now) {
+        /* Why not create a timer with timeout set to DBL_MAX?  See
+         * deadlock error notes in flux_future_wait_for() */
```

Comment:
This comment seems like it's warding off something that would not be intuitive to do.

However, if you think it helps, then I'm fine with leaving it.

Reply:
yeah, i was debating if its better or worse without it.  i'll remove it since you mentioned it :-)